Assets/MRTK/Services/DiagnosticsSystem/MixedRealityToolkitVisualProfiler.cs

```diff
@@ -805,23 +846,38 @@ private static ulong AppMemoryUsageLimit
             }
         }
 
+        private static readonly ProfilerMarker WillDisplayedMemoryUsageDifferPerfMarker = new ProfilerMarker("[MRTK] MixedRealityToolkitVisualProfiler.WillDisplayedMemoryUsageDiffer");
+
         private static bool WillDisplayedMemoryUsageDiffer(ulong oldUsage, ulong newUsage, int displayedDecimalDigits)
         {
-            float oldUsageMBs = ConvertBytesToMegabytes(oldUsage);
-            float newUsageMBs = ConvertBytesToMegabytes(newUsage);
-            float decimalPower = Mathf.Pow(10.0f, displayedDecimalDigits);
+            using (WillDisplayedMemoryUsageDifferPerfMarker.Auto())
+            {
+                float oldUsageMBs = ConvertBytesToMegabytes(oldUsage);
+                float newUsageMBs = ConvertBytesToMegabytes(newUsage);
+                float decimalPower = Mathf.Pow(10.0f, displayedDecimalDigits);
 
-            return (int)(oldUsageMBs * decimalPower) != (int)(newUsageMBs * decimalPower);
+                return (int)(oldUsageMBs * decimalPower) != (int)(newUsageMBs * decimalPower);
+            }
         }
 
+        private static readonly ProfilerMarker ConvertMegabytesToBytesPerfMarker = new ProfilerMarker("[MRTK] MixedRealityToolkitVisualProfiler.ConvertMegabytesToBytes");
+
         private static ulong ConvertMegabytesToBytes(int megabytes)
         {
-            return ((ulong)megabytes * 1024UL) * 1024UL;
+            using (ConvertMegabytesToBytesPerfMarker.Auto())
```

Comment:
nit: Do we need markers this granular?

Reply:
Yeah some of these feel over instrumented (i.e. especially instrumenting itoa())

Reply:
i'll remove

Reply:
done